scripts/zones/Maze_of_Shakhrami/mobs/Leech_King.lua

```diff
@@ -0,0 +1,23 @@
+-----------------------------------
+-- Area: Maze of Shakhrami
+--  MOB: Leech King (NM)
+-----------------------------------
+
+require("scripts/zones/Maze_of_Shakhrami/MobIDs");
+
+-----------------------------------
+-- onMobDeath
+-----------------------------------
+
+function onMobDeath(mob,killer,ally)
+    local whichNM = math.random(0,19);
+
+    if (whichNM < 10) then
+        SetRespawnTime(Argus, 64800, 108000); -- 18-30 hours
+        DeterMob(Leech_King, true);
+    else
+        SetRespawnTime(Leech_King, 64800, 108000); -- 18-30 hours
+        DeterMob(Argus, true);
```

Comment:
But onZoneInitialize is only called once, so if you have more than one kill you need to deter the other mob. Or am I off on this?